app/models/irregular_income_payment.rb

```diff
@@ -0,0 +1,11 @@
+class IrregularIncomePayment < ApplicationRecord
+  ANNUAL_FREQUENCY = 'annual'.freeze
+  STUDENT_LOAN = 'student_loan'.freeze
+  VALID_IRREGULAR_INCOME_FREQUENCIES = [ANNUAL_FREQUENCY].freeze
+  VALID_IRREGULAR_INCOME_TYPES = [STUDENT_LOAN].freeze
+
+  belongs_to :gross_income_summary
+
+  validates_uniqueness_of :income_type
+  validates :amount, numericality: { greater_than_or_equal_to: 0 }
+end
```

Comment:
You still have validates uniqueness of income type here, which means you will only ever be able to have one record in the database where income type is 'student_loan'.

let's just pair on this little bit straight after standup.